example/zod/schemas.ts

```diff
@@ -11,108 +11,84 @@ export const isDefinedNonNullAny = (v: any): v is definedNonNullAny => v !== und
 
 export const definedNonNullAnySchema = z.any().refine((v) => isDefinedNonNullAny(v));
 
-export function AdminSchema(): z.ZodObject<Properties<Admin>> {
-  return z.object<Properties<Admin>>({
-    __typename: z.literal('Admin').optional(),
-    lastModifiedAt: definedNonNullAnySchema.nullish()
-  })
-}
+export const PageTypeSchema = z.nativeEnum(PageType);
 
-export function AttributeInputSchema(): z.ZodObject<Properties<AttributeInput>> {
-  return z.object<Properties<AttributeInput>>({
-    key: z.string().nullish(),
-    val: z.string().nullish()
-  })
-}
+export const HttpMethodSchema = z.nativeEnum(HttpMethod);
+
+export const EventOptionTypeSchema = z.nativeEnum(EventOptionType);
 
 export const ButtonComponentTypeSchema = z.nativeEnum(ButtonComponentType);
 
-export function ComponentInputSchema(): z.ZodObject<Properties<ComponentInput>> {
-  return z.object<Properties<ComponentInput>>({
-    child: z.lazy(() => ComponentInputSchema().nullish()),
-    childrens: z.array(z.lazy(() => ComponentInputSchema().nullable())).nullish(),
-    event: z.lazy(() => EventInputSchema().nullish()),
-    name: z.string(),
-    type: ButtonComponentTypeSchema
-  })
-}
-
-export function DropDownComponentInputSchema(): z.ZodObject<Properties<DropDownComponentInput>> {
-  return z.object<Properties<DropDownComponentInput>>({
-    dropdownComponent: z.lazy(() => ComponentInputSchema().nullish()),
-    getEvent: z.lazy(() => EventInputSchema())
-  })
-}
-
-export function EventArgumentInputSchema(): z.ZodObject<Properties<EventArgumentInput>> {
-  return z.object<Properties<EventArgumentInput>>({
+export const HttpInputSchema: z.ZodObject<Properties<HttpInput>> = z.object({
+    method: HttpMethodSchema.nullish(),
+    url: definedNonNullAnySchema
+});
+
+export const EventArgumentInputSchema: z.ZodObject<Properties<EventArgumentInput>> = z.object({
     name: z.string().min(5),
     value: z.string().regex(/^foo/, "message")
-  })
-}
+});
 
-export function EventInputSchema(): z.ZodObject<Properties<EventInput>> {
-  return z.object<Properties<EventInput>>({
-    arguments: z.array(z.lazy(() => EventArgumentInputSchema())),
+export const EventInputSchema: z.ZodObject<Properties<EventInput>> = z.object({
```

Comment:
is there a reason why this is no longer passing `Properties<EventInput>` to `z.object`? 

I now get errors of the form
```
Type 'ZodObject<{ deviceName: ZodEffects<ZodAny, any, any>; filePath: ZodEffects<ZodAny, any, any>; }, "strip", ZodTypeAny, { deviceName?: any; filePath?: any; }, { ...; }>' is not assignable to type 'ZodObject<Required<{ deviceName: ZodType<string, any, string>; filePath: ZodType<string, any, string>; }>, UnknownKeysParam, ZodTypeAny, { ...; }, { ...; }>'.
  Type '{ deviceName?: any; filePath?: any; }' is not assignable to type '{ deviceName: string; filePath: string; }'.
```
which are fixed if I provide the correct type to `object`.